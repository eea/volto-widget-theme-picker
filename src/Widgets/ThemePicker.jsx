import React from 'react';
import { Form } from 'semantic-ui-react';
import { Grid, Button } from 'semantic-ui-react';

const ThemePicker = (props) => {
  const { id, title, required, value, onChange, colors, className } = props;

  React.useEffect(() => {
    if (!props.value && props.default) {
      props.onChange(props.id, props.default);
    }
  });

  return colors && colors.length > 0 ? (
    <Form.Field
      inline
      required={required}
      className={className}
      id={'field-' + id}
    >
      <Grid>
        <Grid.Row>
          <Grid.Column
            width="12"
            className="color-picker-widget"
            verticalAlign="middle"
          >
            <div className="wrapper">
              <label htmlFor={`field-${id}`}>{title ? title : 'Color'}</label>

              <div className="buttons">
                {colors.map((color) => {
                  if (color.type === 'pluggable') {
                    return (
                      <Button
                        key={id + color.name}
                        className={color.name}
                        onClick={(e) => {
                          e.preventDefault();
                          e.stopPropagation();
                          onChange(id, color.name);
                        }}
                        active={value === color.name}
                        circular
                        aria-label={color.label}
                        title={color.label}
                      />
                    );
                  } else {
                    return (
                      <Button
                        key={id + color.name}
                        style={{ backgroundColor: color.name }}
                        onClick={(e) => {
                          e.preventDefault();
                          e.stopPropagation();
                          onChange(id, color.name);
                        }}
                        active={value === color.name}
                        circular
                        aria-label={color.label}
                        title={'Color: ' + color.label}
                      />
                    );
                  }
                })}
              </div>
            </div>
          </Grid.Column>
        </Grid.Row>
      </Grid>
    </Form.Field>
  ) : null;
};

export default ThemePicker;
