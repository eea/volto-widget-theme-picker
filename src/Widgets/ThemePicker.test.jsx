import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import ThemePicker from './ThemePicker';

describe('ThemePicker', () => {
  const props = {
    id: 'test-theme-picker',
    title: 'Test Theme Picker',
    value: 'color1',
    colors: [
      { name: 'color1', label: 'Color 1' },
      { name: 'color2', label: 'Color 2' },
    ],
    onChange: () => {},
  };

  it('renders correctly', () => {
    const { container } = render(<ThemePicker {...props} />);

    expect(container.querySelector('.wrapper label')).toBeInTheDocument();
    expect(
      container.querySelector('.wrapper .buttons .color1[title="Color 1"]'),
    ).toBeInTheDocument();
    expect(
      container.querySelector('.wrapper .buttons .color2[title="Color 2"]'),
    ).toBeInTheDocument();
  });

  it('calls onChange with correct value when a color button is clicked', () => {
    const onChange = jest.fn();
    const { getByTitle } = render(
      <ThemePicker {...props} onChange={onChange} />,
    );

    fireEvent.click(getByTitle('Color 2'));
    expect(onChange).toHaveBeenCalledWith('test-theme-picker', 'color2');
  });

  it('does not render when no colors are provided', () => {
    const { container } = render(<ThemePicker {...props} colors={[]} />);

    expect(
      container.querySelector('.wrapper .field-test-theme-picker'),
    ).not.toBeInTheDocument();
  });
});
