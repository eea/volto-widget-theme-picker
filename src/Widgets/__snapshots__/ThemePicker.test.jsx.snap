// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemePicker does not render when no colors are provided 1`] = `null`;

exports[`ThemePicker renders correctly 1`] = `
<div
  className="inline field"
  id="field-test-theme-picker"
>
  <div
    className="ui grid"
  >
    <div
      className="row"
    >
      <div
        className="middle aligned twelve wide column color-picker-widget"
      >
        <div
          className="wrapper"
        >
          <label
            htmlFor="field-test-theme-picker"
          >
            Test Theme Picker
          </label>
          <div
            className="buttons"
          >
            <button
              aria-label="Color 1"
              className="ui active circular button color1"
              onClick={[Function]}
              title="Color 1"
            />
            <button
              aria-label="Color 2"
              className="ui circular button color2"
              onClick={[Function]}
              title="Color 2"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
